query-engine/connectors/mongodb-query-connector/src/join.rs

```diff
@@ -134,10 +146,22 @@ impl JoinStage {
         // };
         let unwind_stage = None;
 
+        // Time to deal with the left side of the equation
+        let left_scalars = from_field.left_scalars();
+
+        // With the left side, we need to replace the `left_X` for the right field
+        let mut let_vars = Document::new();
+        for (idx, left_field) in left_scalars.iter().enumerate() {
+            let left_name = format!("left_{}", idx);
+
+            let_vars.insert(left_name, format!("${}", left_field.db_name()));
+        }
```

Comment:
neat pick: we usually put blank lines _before_ and _after_ block expressions (expressions with brackets `{ }`)

Reply:
Also, I know I'm the one who suggested `left_{idx}` as alias for left fields, but `left_{field_name}` might be better to understand more easily the queries.

Reply:
I will prefer and keep the `left_0` style, they are variables and designed to be read by machines, using the field name could complicate the things a little more when reading the query.